Classes/Controller/BackendServiceController.php

```diff
@@ -75,9 +80,64 @@ public function initializeUpdateAction(): void
      *
      * @return void
      */
-    public function updateAction(array $resultDtos): void
+    public function updateAction(array $resultDtos): string
     {
         $this->assetService->updateMultipleAssets($resultDtos);
-        $this->view->assign('value', $resultDtos);
+        return json_encode(array_map(fn (AssetModuleResultDto $item) => $item->jsonSerialize(), $resultDtos));
+    }
+
+    public function initializeGetFocusKeywordNodesAction(): void
+    {
+        try {
+            $propertyMappingConfiguration = $this->arguments->getArgument('configuration')
+                ->getPropertyMappingConfiguration();
+            $propertyMappingConfiguration->allowProperties('workspace');
+            $propertyMappingConfiguration->allowProperties('generateEmptyFocusKeywords');
+            $propertyMappingConfiguration->allowProperties('regenerateExistingFocusKeywords');
+            $propertyMappingConfiguration->allowProperties('nodeTypeFilter');
+            $propertyMappingConfiguration->allowProperties('limit');
+        } catch (NoSuchArgumentException) {
+            // This cannot happen, otherwise we have a broken
+            // request anyway
+        }
+    }
+
+    /**
+     * @param FocusKeywordModuleConfigurationDto $configuration
+     *
+     * @return string|bool
+     */
+    public function getFocusKeywordNodesAction(FocusKeywordModuleConfigurationDto $configuration): string|bool
+    {
+        $resultCollection = $this->nodeService->getNodesThatNeedProcessing($configuration, $this->controllerContext);
+        return json_encode($resultCollection);
+    }
+
+    public function initializeUpdateFocusKeywordNodesAction(): void
+    {
+        try {
+            $propertyMappingConfiguration =
+                $this->arguments->getArgument('resultDtos')->getPropertyMappingConfiguration();
+            $propertyMappingConfiguration
+                ->forProperty(PropertyMappingConfiguration::PROPERTY_PATH_PLACEHOLDER)
+                ->skipProperties('generating', 'persisting', 'persisted')
+                ->allowAllProperties();
+        } catch (NoSuchArgumentException $e) {
```

Comment:
Either handle the error or remove the try-catch and let the upper method handle the error